pallets/sp-mvm/src/lib.rs

```diff
@@ -143,7 +145,9 @@ pub mod pallet {
             Ok(result)
         }
 
+        /// SBP M1 Missing comments
         #[pallet::weight(T::GasWeightMapping::gas_to_weight(*gas_limit))]
+        /// SBP M1 How is weight calculation working? Multiple extrinsics with same weight?
```

Comment:
How do you account for the cost of the extrinsic itself?

Reply:
The logic is so:

1. User passes the gas limit during extrinsic execution. 
2. It converts to weight (gas to weight mapping). 
3. We convert weight to fees using standard Substrate functional here.

Probably we should add some static weight for extrinsic itself, but it's not done yet, as we just finished with integration tests and want to play with gas using integration tests, because currently it seems the gas model should be changed a little (i mean costs table, maybe we also need to change way we map gas to weight).

Reply:
Makes sense! Yes, you should definitely consider the weight of the extrinsic itself.